Content.Shared/GameObjects/EntitySystems/SharedGhostSystem.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using Content.Shared.GameObjects.Components.Observer;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.GameObjects.EntitySystems
+{
+    public abstract class SharedGhostSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<SharedGhostComponent, GhostChangeCanReturnToBodyEvent>(OnGhostChangeCanReturnToBody);
+        }
+
+        private void OnGhostChangeCanReturnToBody(EntityUid uid, SharedGhostComponent component, GhostChangeCanReturnToBodyEvent args)
+        {
+            if (component.CanReturnToBody == args.CanReturnToBody)
+            {
+                return;
+            }
+
+            component.CanReturnToBody = args.CanReturnToBody;
+            component.Dirty();
+        }
+    }
+
+    [Serializable, NetSerializable]
+    public class GhostChangeCanReturnToBodyEvent : EntityEventArgs
```

Comment:
Document this?

Reply:
Done in https://github.com/space-wizards/space-station-14/pull/4159/commits/d1d4589647b7c53883d0ac2fcf4b58d7bf5f7131 and https://github.com/space-wizards/space-station-14/pull/4159/commits/0405adf626bddc81a4bcd0f55ad8b7caef1a180b